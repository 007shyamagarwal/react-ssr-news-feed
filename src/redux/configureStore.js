import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import rootReducer from './reducers';
import { composeWithDevTools } from 'redux-devtools-extension';

const localStorageMiddleware = ({ getState }) => {
  return (next) => (action) => {
    const result = next(action);
    localStorage.setItem(
      `applicationState/${getState().pageId}`,
      JSON.stringify(getState())
    );
    return result;
  };
};

const reHydrateStore = (isServer, pageId) => {
  if (isServer) return;
  if (localStorage.getItem(`applicationState/${pageId}`) !== null) {
    return JSON.parse(localStorage.getItem(`applicationState/${pageId}`)); // re-hydrate the store
  }
};

export default function configureStore(preloadedState, isServer) {
  const { pageId } = preloadedState;
  const middlewares = isServer
    ? [thunkMiddleware]
    : [thunkMiddleware, localStorageMiddleware];
  return createStore(
    rootReducer,
    reHydrateStore(isServer, pageId) || preloadedState,
    composeWithDevTools(applyMiddleware(...middlewares))
  );
}
