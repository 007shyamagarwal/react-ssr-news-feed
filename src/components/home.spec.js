import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Home from './home';

describe('Home', () => {
  const initialState = {
    isFetching: false,
    posts: { hits: [] },
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('Home page with empty content', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Home />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Home page with content', () => {
    const hits = [
      {
        author: 'gerbilly',
        num_comments: 572,
        objectID: '19576092',
        points: 1623,
        title: 'You Are Not Google (2017)',
        url: 'https://blog.bradfieldcs.com/you-are-not-google-84912cf44afb',
      },
    ];
    const store = mockStore({ ...initialState, post: { hits } });

    const tree = renderer
      .create(
        <Provider store={store}>
          <Home />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
