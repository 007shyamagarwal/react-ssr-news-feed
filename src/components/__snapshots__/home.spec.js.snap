// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home page with content 1`] = `
Array [
  <table
    bordered={true}
    striped={true}
  >
    <thead
      className="table-heading"
    >
      <tr>
        <th
          className=" comments"
          style={
            Object {
              "textTransform": "capitalize",
            }
          }
        >
          Comments
        </th>
        <th
          className=" votes"
          style={
            Object {
              "textTransform": "capitalize",
            }
          }
        >
          Vote Count
        </th>
        <th
          className=" upvote"
          style={
            Object {
              "textTransform": "capitalize",
            }
          }
        >
          Upvote
        </th>
        <th
          className=" details"
          style={
            Object {
              "textTransform": "capitalize",
            }
          }
        >
          News Details
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
  <div
    className="btton-container"
  >
    <a
      className="organge"
      href="http://localhost/page/1"
    >
      Previous
    </a>
    <a
      className="organge"
      href="http://localhost/page/1"
    >
      Next
    </a>
  </div>,
  <div
    style={
      Object {
        "border": "1px solid orangered",
      }
    }
  >
    <div
      style={
        Object {
          "height": "300px",
          "marginTop": "30px",
          "width": "100%",
        }
      }
    >
      <div>
        Loading Chart
      </div>
    </div>
  </div>,
]
`;

exports[`Home Home page with empty content 1`] = `
Array [
  <table
    bordered={true}
    striped={true}
  >
    <thead
      className="table-heading"
    >
      <tr>
        <th
          className=" comments"
          style={
            Object {
              "textTransform": "capitalize",
            }
          }
        >
          Comments
        </th>
        <th
          className=" votes"
          style={
            Object {
              "textTransform": "capitalize",
            }
          }
        >
          Vote Count
        </th>
        <th
          className=" upvote"
          style={
            Object {
              "textTransform": "capitalize",
            }
          }
        >
          Upvote
        </th>
        <th
          className=" details"
          style={
            Object {
              "textTransform": "capitalize",
            }
          }
        >
          News Details
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
  <div
    className="btton-container"
  >
    <a
      className="organge"
      href="http://localhost/page/1"
    >
      Previous
    </a>
    <a
      className="organge"
      href="http://localhost/page/1"
    >
      Next
    </a>
  </div>,
  <div
    style={
      Object {
        "border": "1px solid orangered",
      }
    }
  >
    <div
      style={
        Object {
          "height": "300px",
          "marginTop": "30px",
          "width": "100%",
        }
      }
    >
      <div>
        Loading Chart
      </div>
    </div>
  </div>,
]
`;
